python-checks/src/main/java/org/sonar/python/checks/RandomSeedCheck.java

```diff
@@ -67,14 +74,47 @@ public void initialize(Context context) {
 
   private void checkEmptySeedCall(SubscriptionContext ctx) {
     CallExpression call = (CallExpression) ctx.syntaxNode();
-    Optional.ofNullable(call.calleeSymbol()).map(Symbol::fullyQualifiedName)
-        .map(SEED_METHODS_TO_CHECK::get)
-        .filter(argName -> isSeedArgumentAbsentOrNone(argName, call.arguments()))
-        .ifPresent(fqn -> ctx.addIssue(call, MESSAGE));
+
+    Optional<Symbol> maybeCalleeSymbol = Optional.ofNullable(call.calleeSymbol());
+
+    maybeCalleeSymbol
+      .map(Symbol::fullyQualifiedName)
+      .map(SEED_METHODS_TO_CHECK::get)
+      .filter(argName -> isArgumentAbsentOrNone(TreeUtils.nthArgumentOrKeyword(0, argName, call.arguments())))
+      .map(arg -> MESSAGE)
+      .or(() -> maybeCalleeSymbol
+        .filter(symbol -> symbol.fullyQualifiedName() != null && symbol.fullyQualifiedName().startsWith(SKLEARN_FQN))
+        .filter(RandomSeedCheck::hasRandomStateParameter)
+        .filter(symbol -> isArgumentAbsentOrNone(TreeUtils.argumentByKeyword(SKLEARN_ARG_NAME, call.arguments())))
+        .map(symbol -> SKLEARN_MESSAGE))
+      .ifPresent(message -> ctx.addIssue(call, message));
+  }
+
+  private static boolean hasRandomStateParameter(Symbol calleeSymbol) {
+    if (calleeSymbol.is(Kind.CLASS)) {
+      List<Symbol> constructorSymbols = ((ClassSymbolImpl) calleeSymbol).declaredMembers()
+        .stream()
+        .filter(s -> "__init__".equals(s.name()))
+        .toList();
+      if (constructorSymbols.size() == 1) {
+        return hasRandomStateParameter(constructorSymbols.get(0));
+      }
+      return false;
+    }
+    if (calleeSymbol.is(Kind.FUNCTION)) {
+      List<FunctionSymbol> symbols = SymbolUtils.getFunctionSymbols(calleeSymbol);
+      if (symbols.size() == 1) {
+        FunctionSymbol symbol = symbols.get(0);
+        return symbol.parameters()
+          .stream()
+          .map(FunctionSymbol.Parameter::name)
+          .anyMatch(SKLEARN_ARG_NAME::equals);
+      }
+    }
+    return false;
   }
 
-  private boolean isSeedArgumentAbsentOrNone(String argName, List<Argument> args) {
-    RegularArgument arg = TreeUtils.nthArgumentOrKeyword(0, argName, args);
+  private boolean isArgumentAbsentOrNone(@Nullable RegularArgument arg) {
```

Comment:
This is a good candidate to be put in the utils, if it is not there already

Reply:
I will create a ticket for this.